standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -12025,7 +12025,9 @@ private List<SQLPrimaryKey> getPrimaryKeysInternal(final String catName,
                                                      final String db_name_input,
                                                      final String tbl_name_input)
   throws MetaException, NoSuchObjectException {
-    final String db_name = normalizeIdentifier(db_name_input);
+    final String db_name = StringUtils.isNotBlank(db_name_input) ?
```

Comment:
1. can you please fix db_name_input and tbl_name_input naming nightmare as you're touching this part?
2. what about fixing this in normalizeIdentifier instead?
```
return identifier == null ? null : identifier.trim().toLowerCase();
```

Reply:
>what about fixing this in normalizeIdentifier instead?

This backfired 😅 

This method is widely used, has some 500+ usage and some tests explicitly asks for an NPE which later gets converted to MetaException, because of this it doesn't fail here rather in the end fails with NoSuchObjectException in many cases. 

Reverting back to original for now to fix this only

Reply:
oh, this is sooo...hive :)
I don't think we can do anything instantly about this, so my approval still holds